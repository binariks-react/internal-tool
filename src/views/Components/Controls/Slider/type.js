export const SLIDER_TYPE = Object.freeze({
  SINGLE: 'single',
  DOUBLE: 'double',
});

export const CONTROLLER_TYPE = Object.freeze({
  LEFT: 'left-controller',
  RIGHT: 'right-controller',
});
