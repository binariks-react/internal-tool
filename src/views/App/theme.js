export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
