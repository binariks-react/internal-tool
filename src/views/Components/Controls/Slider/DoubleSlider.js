import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Range from './Range';
import Controller from './Controller';
import { CONTROLLER_TYPE } from './type';

const DoubleSlider = (props) => {
  const {
    sliderWidth,
    rangeValue,
    leftControllerValue,
    rightControllerValue,
    minDistanceBetweenControllers,
    rangeColour,
    showValue,
    getRange,
  } = props;

  const convertedOneValueToPixelPresentation = (rangeValue / sliderWidth);

  const getControllerPosition = controllerValue => Math.ceil(controllerValue / convertedOneValueToPixelPresentation);

  // eslint-disable-next-line max-len
  const getMinDistanceBetweenControllers = () => Math.ceil(minDistanceBetweenControllers / convertedOneValueToPixelPresentation);

  const calculateBorderCollision = (controllerType, positionOfNextController) => {
    const minDistanceBetweenControllersInPixels = getMinDistanceBetweenControllers();
    switch (controllerType) {
    case CONTROLLER_TYPE.LEFT: {
      return positionOfNextController - Math.ceil(minDistanceBetweenControllersInPixels / 2);
    }
    case CONTROLLER_TYPE.RIGHT: {
      return positionOfNextController + Math.ceil(minDistanceBetweenControllersInPixels / 2);
    }
    default: {
      throw new Error('We didn\'t get controller type');
    }
    }
  };

  const [leftController, setLeftController] = useState({
    position: getControllerPosition(leftControllerValue),
    minValue: 0,
    maxValue: calculateBorderCollision(CONTROLLER_TYPE.LEFT, getControllerPosition(rightControllerValue)),
  });
  const [rightController, setRightController] = useState({
    position: getControllerPosition(rightControllerValue),
    minValue: calculateBorderCollision(CONTROLLER_TYPE.RIGHT, getControllerPosition(leftControllerValue)),
    maxValue: sliderWidth,
  });

  const handleControllerChange = (controllerPosition, controllerType) => {
    switch (controllerType) {
    case CONTROLLER_TYPE.LEFT: {
      setLeftController({
        ...leftController,
        position: controllerPosition,
        maxValue: calculateBorderCollision(controllerType, rightController.position),
      });
      setRightController({
        ...rightController,
        minValue: calculateBorderCollision(controllerType, controllerPosition),
      });
      break;
    }
    case CONTROLLER_TYPE.RIGHT: {
      setLeftController({
        ...leftController,
        maxValue: calculateBorderCollision(controllerType, controllerPosition),
      });
      setRightController({
        ...rightController,
        position: controllerPosition,
        minValue: calculateBorderCollision(controllerType, leftController.position),
      });
      break;
    }
    default: {
      return new Error('Problem with controllers');
    }
    }
    getRange({
      leftPoint: 0,
      rightPoint: (controllerPosition * convertedOneValueToPixelPresentation).toFixed(2),
    });
  };

  return (
    <SliderWrapper sliderWidth={sliderWidth} showValue={showValue} >
      <Range
        id="tickmarks"
        sliderWidth={sliderWidth}
        leftPoint={leftController.position}
        rightPoint={rightController.position}
        rangeColour={rangeColour}
      />
      <Controller
        initialValue={leftController.position}
        controllerType={CONTROLLER_TYPE.LEFT}
        handleControllerChange={handleControllerChange}
        minValue={leftController.minValue}
        maxValue={leftController.maxValue}
        showValue={showValue}
        sliderWidth={sliderWidth}
        rangeValue={rangeValue}
      />
      <Controller
        initialValue={rightController.position}
        controllerType={CONTROLLER_TYPE.RIGHT}
        handleControllerChange={handleControllerChange}
        minValue={rightController.minValue}
        maxValue={rightController.maxValue}
        showValue={showValue}
        sliderWidth={sliderWidth}
        rangeValue={rangeValue}
      />

    </SliderWrapper>
  );
};

export default DoubleSlider;

DoubleSlider.defaultProps = {};

DoubleSlider.propTypes = {
  sliderWidth: PropTypes.number.isRequired,
  rangeValue: PropTypes.number.isRequired,
  leftControllerValue: PropTypes.number.isRequired,
  rightControllerValue: PropTypes.number.isRequired,
  minDistanceBetweenControllers: PropTypes.number.isRequired,
  rangeColour: PropTypes.string.isRequired,
  getRange: PropTypes.func.isRequired,
  showValue: PropTypes.bool.isRequired,
};

const SliderWrapper = styled.div`
  position: relative;
  margin: ${ props => props.showValue ? '55px 0 35px' : '35px 0'};
  width: ${ props => props.sliderWidth}px;
`;
