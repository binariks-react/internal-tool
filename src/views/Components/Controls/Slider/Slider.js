import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import SingleSlider from './SingleSlider';
import DoubleSlider from './DoubleSlider';

import { SLIDER_TYPE } from './type';

const Slider = props => (
  <React.Fragment>
    {props.type === SLIDER_TYPE.SINGLE &&
      <SingleSlider
        {...props}
        controllerValue={
          props.controllerValue >= 0 && props.controllerValue <= props.rangeValue ?
            props.controllerValue : props.controllerValue > props.rangeValue ? props.rangeValue : 0
        }
      />
    }
    {props.type === SLIDER_TYPE.DOUBLE &&
      <DoubleSlider
        {...props}
        leftControllerValue={props.leftControllerValue >= 0 ? props.leftControllerValue : 0}
        rightControllerValue={props.rightControllerValue <= props.rangeValue ?
          props.rightControllerValue :
          props.rangeValue}
      />
    }
    {
      props.type !== SLIDER_TYPE.SINGLE &&
      props.type !== SLIDER_TYPE.DOUBLE &&
      (<ErrorMessage>You didn't pass normal type of range</ErrorMessage>)
    }
  </React.Fragment>
);

Slider.defaultProps = {
  rangeColour: 'primary',
  showValue: false,
};

Slider.propTypes = {
  sliderWidth: PropTypes.number.isRequired,
  rangeValue: PropTypes.number.isRequired,
  type: PropTypes.oneOf([SLIDER_TYPE.SINGLE, SLIDER_TYPE.DOUBLE]).isRequired,
  // eslint-disable-next-line react/require-default-props
  controllerValue: PropTypes.number,
  // eslint-disable-next-line react/require-default-props
  leftControllerValue: PropTypes.number,
  // eslint-disable-next-line react/require-default-props
  rightControllerValue: PropTypes.number,
  // eslint-disable-next-line react/require-default-props
  minDistanceBetweenControllers: PropTypes.number,
  rangeColour: PropTypes.string,
  getRange: PropTypes.func.isRequired,
  showValue: PropTypes.bool,
};

export default Slider;

const ErrorMessage = styled.div``;
