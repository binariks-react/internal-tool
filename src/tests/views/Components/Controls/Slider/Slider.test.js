import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from 'enzyme';
import theme from 'views/App/theme';
import Slider from 'views/Components/Controls/Slider/Slider';
import { SLIDER_TYPE } from 'views/Components/Controls/Slider/type';
import 'jest-styled-components';

describe('<Slider />', () => {
  const fakeProps = {
    sliderWidth: 100,
    rangeValue: 100,
    type: SLIDER_TYPE.SINGLE,
    controllerValue: 50,
    leftControllerValue: 10,
    rightControllerValue: 50,
    minDistanceBetweenControllers: 10,
    rangeColour: 'primary',
    getRange: () => { },
    showValue: false,
  };
  it('snapshot with one controller without showing value', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('With one controller with showing value', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} showValue />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('With two controllers without showing value', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} type={SLIDER_TYPE.DOUBLE} />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('With two controllers with showing value', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} type={SLIDER_TYPE.DOUBLE} showValue />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('position more than 0, and max range value', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} showValue controllerValue={-10} />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('left controller more or equal than 0 and right controller less than', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} showValue leftControllerValue={-10} />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('right controller less than slider size', () => {
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} showValue rightControllerValue={200} />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('show error message if did not pass slider type', () => {
    const cle = console.error;
    console.error = jest.fn();
    const wrapper = render(
      <ThemeProvider theme={theme}>
        <Slider {...fakeProps} type={null} />
      </ThemeProvider>
    );
    expect(console.error).toHaveBeenCalledTimes(1);
    console.error = cle;
    expect(wrapper).toMatchSnapshot();
  });
});
