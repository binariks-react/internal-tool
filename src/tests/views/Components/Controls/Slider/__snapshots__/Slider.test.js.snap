// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> With one controller with showing value 1`] = `
.c1 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90,97,105,0.1);
}

.c2 {
  position: absolute;
  top: 0;
  left: 0px;
  right: 45px;
  height: 5px;
  border-radius: 5px;
  background-color: #007bff;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(50px);
  -ms-transform: translateX(50px);
  transform: translateX(50px);
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c4:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c5 {
  position: absolute;
  top: -150%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  margin: 0;
  line-height: 15px;
  color: #5a6169;
  background-color: #fff;
  font-weight: 300;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  border-radius: 5px;
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: 250ms;
  transition: 250ms;
  font-size: 14px;
  padding: 5px 10px;
}

.c0 {
  position: relative;
  margin: 55px 0 35px;
  width: 100px;
}

<div
  class="c0"
  data-test="single-slider"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    >
      <div
        class="c5"
        data-test="controller-text"
      >
        50.00
      </div>
    </div>
  </div>
</div>
`;

exports[`<Slider /> With two controllers with showing value 1`] = `
.c1 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90,97,105,0.1);
}

.c2 {
  position: absolute;
  top: 0;
  left: 10px;
  right: 45px;
  height: 5px;
  border-radius: 5px;
  background-color: #007bff;
}

.c6 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(50px);
  -ms-transform: translateX(50px);
  transform: translateX(50px);
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(10px);
  -ms-transform: translateX(10px);
  transform: translateX(10px);
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c4:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c5 {
  position: absolute;
  top: -150%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  margin: 0;
  line-height: 15px;
  color: #5a6169;
  background-color: #fff;
  font-weight: 300;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  border-radius: 5px;
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: 250ms;
  transition: 250ms;
  font-size: 14px;
  padding: 5px 10px;
}

.c0 {
  position: relative;
  margin: 55px 0 35px;
  width: 100px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    >
      <div
        class="c5"
        data-test="controller-text"
      >
        10.00
      </div>
    </div>
  </div>
  <div
    class="c6"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    >
      <div
        class="c5"
        data-test="controller-text"
      >
        50.00
      </div>
    </div>
  </div>
</div>
`;

exports[`<Slider /> With two controllers without showing value 1`] = `
.c1 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90,97,105,0.1);
}

.c2 {
  position: absolute;
  top: 0;
  left: 10px;
  right: 45px;
  height: 5px;
  border-radius: 5px;
  background-color: #007bff;
}

.c5 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(50px);
  -ms-transform: translateX(50px);
  transform: translateX(50px);
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(10px);
  -ms-transform: translateX(10px);
  transform: translateX(10px);
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c4:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c0 {
  position: relative;
  margin: 35px 0;
  width: 100px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    />
  </div>
  <div
    class="c5"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`<Slider /> left controller more or equal than 0 and right controller less than 1`] = `
.c1 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90,97,105,0.1);
}

.c2 {
  position: absolute;
  top: 0;
  left: 0px;
  right: 45px;
  height: 5px;
  border-radius: 5px;
  background-color: #007bff;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(50px);
  -ms-transform: translateX(50px);
  transform: translateX(50px);
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c4:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c5 {
  position: absolute;
  top: -150%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  margin: 0;
  line-height: 15px;
  color: #5a6169;
  background-color: #fff;
  font-weight: 300;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  border-radius: 5px;
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: 250ms;
  transition: 250ms;
  font-size: 14px;
  padding: 5px 10px;
}

.c0 {
  position: relative;
  margin: 55px 0 35px;
  width: 100px;
}

<div
  class="c0"
  data-test="single-slider"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    >
      <div
        class="c5"
        data-test="controller-text"
      >
        50.00
      </div>
    </div>
  </div>
</div>
`;

exports[`<Slider /> position more than 0, and max range value 1`] = `
.c1 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90,97,105,0.1);
}

.c2 {
  position: absolute;
  top: 0;
  left: 0px;
  right: 95px;
  height: 5px;
  border-radius: 5px;
  background-color: #007bff;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c4:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c5 {
  position: absolute;
  top: -150%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  margin: 0;
  line-height: 15px;
  color: #5a6169;
  background-color: #fff;
  font-weight: 300;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  border-radius: 5px;
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: 250ms;
  transition: 250ms;
  font-size: 14px;
  padding: 5px 10px;
}

.c0 {
  position: relative;
  margin: 55px 0 35px;
  width: 100px;
}

<div
  class="c0"
  data-test="single-slider"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    >
      <div
        class="c5"
        data-test="controller-text"
      >
        0.00
      </div>
    </div>
  </div>
</div>
`;

exports[`<Slider /> right controller less than slider size 1`] = `
.c1 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90,97,105,0.1);
}

.c2 {
  position: absolute;
  top: 0;
  left: 0px;
  right: 45px;
  height: 5px;
  border-radius: 5px;
  background-color: #007bff;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(50px);
  -ms-transform: translateX(50px);
  transform: translateX(50px);
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c4:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c5 {
  position: absolute;
  top: -150%;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  margin: 0;
  line-height: 15px;
  color: #5a6169;
  background-color: #fff;
  font-weight: 300;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  border-radius: 5px;
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: 250ms;
  transition: 250ms;
  font-size: 14px;
  padding: 5px 10px;
}

.c0 {
  position: relative;
  margin: 55px 0 35px;
  width: 100px;
}

<div
  class="c0"
  data-test="single-slider"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    >
      <div
        class="c5"
        data-test="controller-text"
      >
        50.00
      </div>
    </div>
  </div>
</div>
`;

exports[`<Slider /> show error message if did not pass slider type 1`] = `
<div
  class=""
>
  You didn't pass normal type of range
</div>
`;

exports[`<Slider /> snapshot with one controller without showing value 1`] = `
.c1 {
  position: relative;
  box-sizing: border-box;
  width: 100%;
  height: 5px;
  background: #eceeef;
  border-radius: 5px;
  box-shadow: inset 0 1px 2px rgba(90,97,105,0.1);
}

.c2 {
  position: absolute;
  top: 0;
  left: 0px;
  right: 45px;
  height: 5px;
  border-radius: 5px;
  background-color: #007bff;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateX(50px);
  -ms-transform: translateX(50px);
  transform: translateX(50px);
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 21px;
  height: 21px;
  border-radius: 50%;
  background-color: #fff;
  border: 1px solid #e7e9ea;
  -webkit-transition: 250ms;
  transition: 250ms;
  width: 21px;
  height: 21px;
  -webkit-transform: translate(-15%,-8px);
  -ms-transform: translate(-15%,-8px);
  transform: translate(-15%,-8px);
  box-shadow: 0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c4:active {
  cursor: -webkit-grabbing;
  cursor: -moz-grabbing;
  cursor: grabbing;
}

.c4:focus {
  outline: none;
  box-shadow: 0 0 7px rgba(0,123,255,.65),0 3px 15px rgba(90,97,105,.1),0 2px 3px rgba(90,97,105,.2);
}

.c0 {
  position: relative;
  margin: 35px 0;
  width: 100px;
}

<div
  class="c0"
  data-test="single-slider"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-test="controller-wrapper"
  >
    <div
      class="c4"
      data-test="controller-button"
      tabindex="0"
    />
  </div>
</div>
`;
